src/System.Security.Cryptography.Pkcs/src/Internal/Cryptography/Pal/AnyOS/ManagedPal.Decrypt.cs

```diff
@@ -60,11 +61,17 @@ private sealed class ManagedDecryptorPal : DecryptorPal
                     return null;
                 }
 
-                ReadOnlyMemory<byte>? encryptedContent = _envelopedData.EncryptedContentInfo.EncryptedContent;
+                // Pin CEK to prevent heap compaction from copying the data
+                GCHandle cekPin = GCHandle.Alloc(cek, GCHandleType.Pinned);
```

Comment:
Could any line between _GCHandle.Alloc_ and _GCHandle.Free_ throw? If so, there's a potential memory leak here. Generally prefer using the _fixed_ keyword when possible since it's much less susceptible to problems and has orders of magnitude better performance compared to _GCHandle_. I know there are some cases (like the _Encrypt_ function) where you can't use _fixed_.